extensions/applicationinsights-angular-js/src/AngularPlugin.ts

```diff
@@ -30,19 +32,24 @@ export default class AngularPlugin extends BaseTelemetryPlugin {
             }
         });
         if (extConfig.router) {
+            let isPageInitialLoad = true;
+            if (isPageInitialLoad) {
+                const pageViewTelemetry: IPageViewTelemetry = {
+                    uri: extConfig.router.url
+                };
+                this.trackPageView(pageViewTelemetry);
+            }
             extConfig.router.events.subscribe(event => {
-                if (event.constructor.name === "NavigationEnd") {
-                    // Timeout to ensure any changes to the DOM made by route changes get included in pageView telemetry
-                    setTimeout(() => {
```

Comment:
Should this `setTimeout` be kept around?

Reply:
Good question.. I actually don't understand why this setTimeout is required..

Reply:
I mean, reading the doc [here](https://angular.io/api/router/Event) , when router event type is "NavigationEnd" it means "navigation ends successfully", when navigation ends successfully then it's time to trigger track page view isn't it? 🤔

Reply:
I originally added the `setTimeout` in the analytics plugin because the route change events would trigger before the URL itself would change. This caused the pageview telemetry to have the previous URL attached to it, instead of the upcoming one. If the pageviews are correct in your test app, then I think the setTimeout can safely be removed.

Reply:
@markwolff Thanks for bring this up, it's a very good question.. I double checked with my test app and sample app, so the router events happen in this [sequence](https://angular.io/api/router/Event#description). And the `router.url` changes from old url to new url on `ActivationEnd`. And the event listener subscribes to `NavigationEnd` happens after `ActivationEnd` so it catches the new url. 🤓